pkg/config/schema/schema.go

```diff
@@ -0,0 +1,607 @@
+	External ResolveServiceHostname `yaml:"external,omitempty" json:"external,omitempty"`
+}
+
+type ResolveServiceService struct {
+	Service string `yaml:"service,omitempty" json:"service,omitempty"`
+}
+
+type ResolveServiceHostname struct {
+	Hostname string `yaml:"hostname,omitempty" json:"hostname,omitempty"`
+}
+
+type NetworkingAdvanced struct {
+	ClusterDomain string                `yaml:"clusterDomain,omitempty" json:"clusterDomain,omitempty"`
+	FallBack      []NetworkDNSFallback  `yaml:"fallback,omitempty" json:"fallback,omitempty"`
+	ProxyKubelets *NetworkProxyKubelets `yaml:"proxyKubelets,omitempty" json:"proxyKubelets,omitempty"`
+}
+
+type NetworkDNSFallback struct {
+	IP          string `yaml:"ip,omitempty" json:"ip,omitempty"`
+	HostCluster bool   `yaml:"hostCluster,omitempty" json:"hostCluster,omitempty"`
+}
+
+type NetworkProxyKubelets struct {
+	ByHostname bool `yaml:"byHostname,omitempty" json:"byHostname,omitempty"`
+	ByIP       bool `yaml:"byIP,omitempty" json:"byIP,omitempty"`
+}
+
+type Plugins struct {
+	Image  string                 `yaml:"image,omitempty" json:"image,omitempty"`
+	Config map[string]interface{} `yaml:"config,omitempty" json:"config,omitempty"`
+	RBAC   *PluginsRBAC           `yaml:"rbac,omitempty" json:"rbac,omitempty"`
+}
+
+type PluginsRBAC struct {
+	Role        []rbacv1.PolicyRule `yaml:"role,omitempty" json:"role,omitempty"`
+	ClusterRole []rbacv1.PolicyRule `yaml:"clusterRole,omitempty" json:"clusterRole,omitempty"`
+}
+
+type Plugin struct {
+	Plugins `yaml:",inline" json:",inline"`
+
+	Version string `yaml:"version,omitempty" json:"version,omitempty"`
+}
+
+type ControlPlane struct {
+	Distro           Distributions                `yaml:"distro,omitempty" json:"distro,omitempty"`
+	BackingStore     BackingStore                 `yaml:"backingStore,omitempty" json:"backingStore,omitempty"`
+	HostPathMapper   HostPathMapper               `yaml:"hostPathMapper,omitempty" json:"hostPathMapper,omitempty"`
+	CoreDNS          CoreDNS                      `yaml:"coredns,omitempty" json:"coredns,omitempty"`
+	VirtualScheduler *Enablement                  `yaml:"virtualScheduler,omitempty" json:"virtualScheduler,omitempty" product:"pro"`
+	Proxy            *ControlPlaneProxy           `yaml:"proxy,omitempty" json:"proxy,omitempty"`
+	Service          ControlPlaneService          `yaml:"service,omitempty" json:"service,omitempty"`
+	Ingress          ControlPlaneIngress          `yaml:"ingress,omitempty" json:"ingress,omitempty"`
+	HighAvailability ControlPlaneHighAvailability `yaml:"highAvailability,omitempty" json:"highAvailability,omitempty"`
+	Advanced         ControlPlaneAdvanced         `yaml:"advanced,omitempty" json:"advanced,omitempty"`
+}
+
+type Distributions struct {
+	K3S      *DistributionsK3s    `yaml:"k3s,omitempty" json:"k3s,omitempty"`
+	K8S      *DistributionsK8s    `yaml:"k8s,omitempty" json:"k8s,omitempty"`
+	K0S      *corev1.Container    `yaml:"k0s,omitempty" json:"k0s,omitempty"`
```

Comment:
imo this should not be containers because that lets too much freedom and it's going to be super hard to support if everyone can modify all the settings of every container

Reply:
also here, this should probably set under `apiserverimage` to be clear what that sets